configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -248,8 +248,22 @@ else if (!prefix.isEmpty()) {
         }
 
         for (ConstraintViolation<?> violation : VALIDATOR.validate(instance)) {
-            problems.addError("Constraint violation with property prefix '%s': %s %s (for class %s)",
-                    prefix, violation.getPropertyPath(), violation.getMessage(), configClass.getName());
+            String propertyFieldName = violation.getPropertyPath().toString();
+            // upper case first character to match config attribute name
```

Comment:
I don't understand this. Why would anything other than a class name start with uppercase?